archai/nlp/nvidia_transformer_xl/nvidia_utils/cyclic_cosine_scheduler.py

```diff
@@ -0,0 +1,128 @@
+from collections.abc import Iterable
+from math import log, cos, pi, floor
+
+from torch.optim.lr_scheduler import _LRScheduler
+
+
```

Comment:
Could you please add comments on why we need this, where this code comes form and may be line or two of general overview?